backend/infrahub/actions/tasks.py

```diff
@@ -0,0 +1,122 @@
+from __future__ import annotations
+
+from infrahub_sdk.graphql import Mutation
+from prefect import flow
+from prefect.client.orchestration import get_client
+
+from infrahub.context import InfrahubContext  # noqa: TC001  needed for prefect flow
+from infrahub.services import InfrahubServices  # noqa: TC001  needed for prefect flow
+from infrahub.trigger.models import TriggerType
+from infrahub.trigger.setup import setup_triggers
+from infrahub.workflows.catalogue import ACTION_RUN_GENERATOR
+from infrahub.workflows.utils import add_tags
+
+from .gather import gather_trigger_action_rules
+from .models import EventGroupMember  # noqa: TC001  needed for prefect flow
+
+
+@flow(
+    name="action-add-node-to-group",
+    flow_run_name="Adding node={node_id} to group={group_id}",
+)
+async def add_node_to_group(
+    branch_name: str,
+    node_id: str,
+    group_id: str,
+    context: InfrahubContext,  # noqa: ARG001
+    service: InfrahubServices,
+) -> None:
+    await add_tags(branches=[branch_name], nodes=[node_id, group_id])
+
+    mutation = Mutation(
+        mutation="RelationshipAdd",
+        input_data={"data": {"id": group_id, "name": "members", "nodes": [{"id": node_id}]}},
+        query={"ok": None},
+    )
+
+    await service.client.execute_graphql(query=mutation.render(), branch_name=branch_name)
+
+
+@flow(
+    name="action-remove-node-from-group",
+    flow_run_name="Removing node={node_id} from group={group_id}",
+)
+async def remove_node_from_group(
+    branch_name: str,
+    node_id: str,
+    group_id: str,
+    context: InfrahubContext,  # noqa: ARG001
+    service: InfrahubServices,
+) -> None:
+    await add_tags(branches=[branch_name], nodes=[node_id, group_id])
+
+    mutation = Mutation(
+        mutation="RelationshipRemove",
+        input_data={"data": {"id": group_id, "name": "members", "nodes": [{"id": node_id}]}},
+        query={"ok": None},
+    )
+
+    await service.client.execute_graphql(query=mutation.render(), branch_name=branch_name)
+
+
+@flow(
+    name="action-run-generator",
+    flow_run_name="Running generator generator_definition_id={generator_definition_id} on branch={branch_name} for nodes={node_ids}",
+)
+async def run_generator(
+    branch_name: str,
+    node_ids: list[str],
+    generator_definition_id: str,
+    context: InfrahubContext,  # noqa: ARG001
+    service: InfrahubServices,
+) -> None:
+    await add_tags(branches=[branch_name], nodes=node_ids + [generator_definition_id])
+
+    mutation = Mutation(
+        mutation="CoreGeneratorDefinitionRun",
+        input_data={"data": {"id": generator_definition_id, "nodes": node_ids}},
+        query={"ok": None},
+    )
+
+    await service.client.execute_graphql(query=mutation.render(), branch_name=branch_name)
+
+
+@flow(
+    name="action-run-generator-group-event",
+    flow_run_name="Running generator",
```

Comment:
I think we need to revisit this pattern where a flow start another flow and exit immediately, users have reported that it's confusing and hard to 
here if we have only one flow to execute, I think we should call the function directly instead of using `submit_workflow`

Reply:
I had some issues executing the flow decorated function from another flow. What i did was to break out the code into a separate function and then use that one from both flows.